app/src/org/commcare/activities/FormEntryActivityUIController.java

```diff
@@ -694,7 +696,13 @@ protected void hideCompoundIntentCalloutButton() {
         activity.findViewById(R.id.multiple_intent_dispatch_button).setVisibility(View.GONE);
     }
 
-    protected void updateFormRelevancies() {
+    protected synchronized void updateFormRelevancies() {
+        if (formRelevanciesUpdateInProgress) {
```

Comment:
Hey, really glad you caught that, the `synchronized` keyword shouldn't be there anymore. That was leftover from debugging attempts when I thought that the duplicate calls might have been coming from another thread, which turned out not to be the case. Removing it now!